src/community/community.dtos.ts

```diff
@@ -0,0 +1,252 @@
+import {
+    IsOptional,
+    IsArray,
+    IsString,
+    MaxLength,
+    IsNumberString,
+    IsMongoId,
+    IsNotEmpty,
+    ValidateNested, IsIn, IsEthereumAddress, ArrayNotEmpty
+} from 'class-validator';
+import { ApiProperty, OmitType, PartialType, IntersectionType } from '@nestjs/swagger';
+import { Type } from 'class-transformer';
+import { ObjectId } from 'mongodb';
+import constants from '../common/constants';
+
+
+export abstract class CommunityAdministratorBase {
+    @IsNotEmpty()
+    @IsIn(Array.from(constants.PROVIDERS.keys()))
```

Comment:
what is PROVIDERS?

Reply:
That is just a lookup of 'auth' providers but they also overlap with clients so it is used for both.